File: MaxMin/sol.java
import java.io.*;
import java.math.*;
//import java.security.*;
//import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class sol
{

    // Complete the maxMin function below.
    static int maxMin(int k, int[] arr)
    {
        Arrays.sort(arr);
        ArrayList<Integer> results = new ArrayList<Integer>();
        ArrayList<Integer> chunks = new ArrayList<Integer>();
        for(int i = 0; i < k; i++)
        {
            chunks.add(arr[i]);
        }
        for(int i = 0; i < (arr.length - k); i++)
        {
            results.add(chunks.get(k-1) - chunks.get(0));
            chunks.remove(0);
            chunks.add(arr[i+k]);
        }
        results.add(chunks.get(k-1) - chunks.get(0));
        return Collections.min(results);
    }

    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) throws IOException
    {
        //BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));
        int n = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");
        int k = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");
        int[] arr = new int[n];
        for (int i = 0; i < n; i++)
        {
            int arrItem = scanner.nextInt();
            scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");
            arr[i] = arrItem;
        }
        int result = maxMin(k, arr);
        /*bufferedWriter.write(String.valueOf(result));
        bufferedWriter.newLine();
        bufferedWriter.close();*/
        System.out.println(result);
        scanner.close();
    }
}

File: MaxMin/sol.py
#!/bin/python3

import math
#import os
#import random
import re
import sys

# Complete the maxMin function below.
def maxMin(k, arr):
    #Brute Strength
    arr.sort()                                                          #We sort the array. This will make everything else so much easier. 
    #print(arr)
    results = []                                                        #The temporal results will be stored in results. we could instead calculate the minimum with an integer, but, we would need a comparison for every operation. might be too slow. 
    chunks = []                                                         #Chunks will store the temporal arrays the size of k.
    for j in range(k):
        chunks.append(arr[j])                                           #The first chunk of size k is created. then, we can simply delete the first element (chunks.pop(0)) and append the next element (chunks.append())
    #print(chunks)
    for i in range(len(arr)-k):
        #print("Current case: {}".format(chunks[k-1] - chunks[0]))
        results.append(chunks[k-1] - chunks[0])                         #Append the temporal result to the results array
        #results.append(max(chunks) - min(chunks))
        chunks.pop(0)
        chunks.append(arr[i + k])
        #print(chunks)
    results.append(chunks[k-1] - chunks[0])                             #One last result, to consider the final chunk
    return min(results)                                                 #We return the smallest of the results. 

if __name__ == '__main__':
    #fptr = open(os.environ['OUTPUT_PATH'], 'w')
    n = int(input())
    k = int(input())
    arr = []
    for _ in range(n):
        arr_item = int(input())
        arr.append(arr_item)
    result = maxMin(k, arr)
    print(result)
    #fptr.write(str(result) + '\n')
    #fptr.close()